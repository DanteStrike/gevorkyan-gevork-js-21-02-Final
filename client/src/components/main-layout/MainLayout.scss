@import "src/assets/mixins";

.main-layout {
  display: grid;
  grid-template-areas: "header" "content" "footer";
  grid-template-rows: auto 1fr auto;
  grid-template-columns: 1fr;
  grid-gap: 15px;
  height: 100%;

  &__header {
    grid-area: header;
  }

  &__content {
    @include container;
    display: flex;
    grid-area: content;
    align-items: center;
    justify-content: center;
  }

  &__footer {
    grid-area: footer;
  }
}
