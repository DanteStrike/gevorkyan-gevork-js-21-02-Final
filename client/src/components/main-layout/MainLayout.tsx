import React from 'react';
import './MainLayout.scss';
import Header from '../header/Header';
import Routes from '../routes/Routes';
import Footer from '../footer/Footer';

function MainLayout() {
  return (
    <div className="main-layout">
      <div className="main-layout__header">
        <Header />
      </div>
      <div className="main-layout__content">
        <Routes />
      </div>
      <div className="main-layout__footer">
        <Footer />
      </div>
    </div>
  );
}

export default React.memo(MainLayout);
