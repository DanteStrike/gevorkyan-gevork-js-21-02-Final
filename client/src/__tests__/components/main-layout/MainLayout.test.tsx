import {shallow} from 'enzyme';
import React from 'react';
import MainLayout from '../../../components/main-layout/MainLayout';

describe(`Component MainLayout should render correctly`, () => {
  it(`snapshot default test`, () => {
    const wrapper = shallow(<MainLayout />);
    expect(wrapper).toMatchSnapshot();
  });
});
