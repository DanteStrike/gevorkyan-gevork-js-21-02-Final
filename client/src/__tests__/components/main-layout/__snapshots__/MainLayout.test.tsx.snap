// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component MainLayout should render correctly snapshot default test 1`] = `
<div
  className="main-layout"
>
  <div
    className="main-layout__header"
  >
    <Memo(Header) />
  </div>
  <div
    className="main-layout__content"
  >
    <Memo(Routes) />
  </div>
  <div
    className="main-layout__footer"
  >
    <Memo(Footer) />
  </div>
</div>
`;
